src/wp-includes/rest-api.php

```diff
@@ -898,7 +898,7 @@ function rest_filter_response_fields( $response, $server, $request ) {
 
 	$fields = wp_parse_list( $request['_fields'] );
 
-	if ( 0 === count( $fields ) ) {
+	if ( 0 === count( $fields ) || is_scalar( $data ) ) {
```

Comment:
Is scalar desired or valid? If no, thinking it should also trigger an notice (doing it wrong or `wp_trigger_error()`) to alert the devs.

Reply:
A scalar is a valid return type from a REST API controller, so I don't think we need to warn on it.